src/main/java/com/commercetools/sync/categories/README.md

```diff
@@ -0,0 +1,227 @@
+  `Throwable` exception. 
+  
+  An example of setting a callback would be for example logging the error as follows:
+  ````java
+  // instantiate a logger
+  final Logger LOGGER = LoggerFactory.getLogger(myclass.class);
+
+  // instantiating a CategorySyncOptions
+  final CategorySyncOptions categorySyncOptions = CategorySyncOptionsBuilder
+                                                    .of(ctpClient)
+                                                    .setErrorCallBack(LOGGER::error)
+                                                    .build();
+  ````
+  
+  #### `warningCallBack` [Optional]
+  Defines an optional field which represents a callback function which will be called whenever an event occurs
+  that leads to an warning alert from the sync process. The function takes one parameter: a `String` error message.
+  An example of setting a callback would be for example logging the error as follows:
+      
+  ````java
+   // instantiate a logger
+   final Logger LOGGER = LoggerFactory.getLogger(myclass.class);
+    
+   // instantiating a CategorySyncOptions
+   final CategorySyncOptions categorySyncOptions = CategorySyncOptionsBuilder
+                                                        .of(ctpClient)
+                                                        .setWarningCallBack(LOGGER::warn)
+                                                        .build();
+   ````
+  
+  #### `removeOtherLocales` [Optional]
+  Defines an optional field which represents a boolean flag which adds additional localizations without deleting
+  existing ones. If set to true, which is the default value of the option, it deletes the existing localizations. 
+  If set to false, it doesn't delete the existing ones.
+  ````java
+  // instantating options which deletes the existing localizations
+   final CategorySyncOptions categorySyncOptions = CategorySyncOptionsBuilder
+                                                        .of(ctpClient)
+                                                        .setRemoveOtherLocales(true)
+                                                        .build();
+  ````
+  #### `removeOtherSetEntries` [Optional]
+  Defines an optional field which represents a boolean flag which adds additional Set entries without deleting
+  existing ones. If set to true, which is the default value of the option, it deletes the existing Set entries. 
+  If set to false, it doesn't delete the existing ones.
+  ````java
+  // instantating options which deletes the existing Set entries
+   final CategorySyncOptions categorySyncOptions = CategorySyncOptionsBuilder
+                                                        .of(ctpClient)
+                                                        .removeOtherSetEntries(true)
+                                                        .build();
+  ````
+  #### `removeOtherCollectionEntries` [Optional]
```

Comment:
ok but please adapt naming and description as it is not clear which localized fields are affected.